x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter, RequestHandlerContext, APICaller } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { SearchResponse } from 'elasticsearch';
+import { createHash } from 'crypto';
+import { i18n } from '@kbn/i18n';
+
+import lzma from 'lzma-native';
+import { WhitelistRule, WhitelistSet } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const whitelistIdx: string = 'whitelist'; // TODO: change this
+const allowlistBaseRoute: string = '/api/endpoint/allowlist';
+const whitelistLimit: number = 1000;
+let whitelistArtifactCache: Buffer = Buffer.from([]);
+let whitelistCount: number = 0;
```

Comment:
I don't see any reason this needs to be global... why not just have `hydrate` return the count and keep it local?

Reply:
We need to store it globally so we can enforce the limit. https://github.com/elastic/kibana/pull/58650/files#diff-79d0d5e73f351fab815a7879e759c859R127